frontend/src/pages/projects/screens/spawner/deploymentSize/ContainerSizeSelector.tsx

```diff
@@ -21,9 +30,20 @@ const ContainerSizeSelector: React.FC<ContainerSizeSelectorProps> = ({
     <FormGroup label="Container size" fieldId="container-size" data-testid="container-size-group">
       <Select
         isOpen={sizeDropdownOpen}
-        onToggle={() => setSizeDropdownOpen(!sizeDropdownOpen)}
-        aria-label="Container size select"
-        selections={value.name}
+        onOpenChange={(open) => setSizeDropdownOpen(open)}
+        popperProps={{ appendTo: getDashboardMainContainer() }}
+        toggle={(toggleRef) => (
+          <MenuToggle
```

Comment:
After choosing an option using the keyboard, the menu closes and my focus is brought back to the page at the very first focusable element on the page. I lose my context in the form and have to tab back to where I previously was. Focus should return to the menu toggle. This seems to happen with other select menus as well such as the env variable select menus.

Look to abstract out common functionality to reusable components.